Parser/pegen/pegen.h

```diff
@@ -158,10 +158,18 @@ CHECK_CALL_NULL_ALLOWED(Parser *p, void *result)
 #define CHECK(result) CHECK_CALL(p, result)
 #define CHECK_NULL_ALLOWED(result) CHECK_CALL_NULL_ALLOWED(p, result)
 
-#define NEW_TYPE_COMMENT(tc) (tc == NULL ? NULL : _PyPegen_new_type_comment(p, PyBytes_AsString(((Token *)tc)->bytes)))
-
 PyObject *_PyPegen_new_type_comment(Parser *, char *);
-arg_ty _PyPegen_add_type_comment_to_arg(Parser *, arg_ty, char *);
+
+Py_LOCAL_INLINE(PyObject *)
+NEW_TYPE_COMMENT(Parser *p, Token *tc)
+{
+    if (tc == NULL) {
+        return NULL;
+    }
+    return CHECK_CALL_NULL_ALLOWED(p, _PyPegen_new_type_comment(p, PyBytes_AsString(tc->bytes)));
```

Comment:
This will segfault if `PyBytes_AsString` fails. I think the best way to do it is assign it to a variable beforehand and check for `NULL`, but that means the the `CHECK_NULL_ALLOWED` call cannot be inside `NEW_TYPE_COMMENT` or it has to be called multiple times, once for each time a function is called that could potentially return `NULL`, which isn't much more beautiful than having it in the grammar IMHO. What do you think?
```suggestion
    char *bytes = PyBytes_AsString(tc->bytes);
    if (!bytes) {
        return NULL;
    }
    return _PyPegen_new_type_comment(p, bytes);
```

Reply:
Okay, I will try again. I really don't want to litter the grammar with `CHECK_CALL_NULL_ALLOWED(NEW_TYPE_COMMENT(p, tc))` so I am changing `NEW_TYPE_COMMENT` to set `p->error_indicator` on every error exit.